packages/strapi-design-system/src/Select/SingleSelect.tsx

```diff
@@ -0,0 +1,330 @@
+                  aria-label={clearLabel}
+                  title={clearLabel}
+                  ref={clearRef}
+                >
+                  <Cross />
+                </IconBox>
+              ) : null}
+              <DownIcon>
+                <CarretDown />
+              </DownIcon>
+            </Flex>
+          </Trigger>
+          <RadixSelect.Portal>
+            <Content position="popper" sideOffset={4}>
+              <RadixSelect.Viewport ref={viewportRef}>
+                {children}
+                <Box id={intersectionId} width="100%" height="1px" />
+              </RadixSelect.Viewport>
+            </Content>
+          </RadixSelect.Portal>
+        </RadixSelect.Root>
+        <FieldHint />
+        <FieldError />
+      </Stack>
+    </Field>
+  );
+};
+
+interface TriggerProps {
+  $hasError: boolean;
+  $size: Required<Pick<SingleSelectProps, 'size'>>['size'];
+}
+
+const Trigger = styled(RadixSelect.Trigger)<TriggerProps>`
+  position: relative;
+  border: 1px solid ${({ theme, $hasError }) => ($hasError ? theme.colors.danger600 : theme.colors.neutral200)};
+  padding-right: ${({ theme }) => theme.spaces[3]};
+  padding-left: ${({ theme }) => theme.spaces[3]};
+  border-radius: ${({ theme }) => theme.borderRadius};
+  background: ${({ theme }) => theme.colors.neutral0};
+  overflow: hidden;
+  min-height: ${({ theme, $size }) => getThemeSize('input')({ theme, size: $size })};
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  gap: ${({ theme }) => theme.spaces[4]};
+
+  [aria-disabled='true'] {
+    color: ${(props) => props.theme.colors.neutral600};
+    background: ${(props) => props.theme.colors.neutral150};
+  }
+
+  ${({ theme, $hasError }) => inputFocusStyle()({ theme, hasError: $hasError })};
+`;
+
+const DownIcon = styled(RadixSelect.Icon)`
+  & > svg {
+    width: 6px;
+
+    & > path {
+      fill: ${({ theme }) => theme.colors.neutral600};
+    }
+  }
+`;
+
+const Content = styled(RadixSelect.Content)`
+  z-index: 4;
+  background: ${({ theme }) => theme.colors.neutral0};
+  box-shadow: ${({ theme }) => theme.shadows.filterShadow};
+  border: 1px solid ${({ theme }) => theme.colors.neutral150};
+  border-radius: ${({ theme }) => theme.borderRadius};
+  padding: ${({ theme }) => theme.spaces[1]};
+  width: var(--radix-select-trigger-width);
+  max-height: 15rem;
+`;
+
+const IconBox = styled(Box)`
+  background: transparent;
+  border: none;
+  position: relative;
+  z-index: 1;
+
+  svg {
+    height: ${11 / 16}rem;
+    width: ${11 / 16}rem;
+  }
+
+  svg path {
+    fill: ${({ theme }) => theme.colors.neutral600};
+  }
+`;
+
+/***
+ *
+ * Option
+ *
+ */
+
+export interface SingleSelectOptionProps {
+  children: string | number;
+  startIcon?: React.ReactNode;
+  value: string | number;
+}
+
+export const SingleSelectOption = ({ children, startIcon, value }: SingleSelectOptionProps) => (
+  <SelectItem data-strapi-value={value} value={value.toString()}>
```

Comment:
We we still need the data attribute?

Reply:
Don't think so, probably a hangover from how the old select worked.